optimism/src/keccak/constraints.rs

```diff
@@ -159,6 +163,76 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
                     }
                 }
             } // END theta
+
+            // STEP pirho: 5 * 5 * (2 + 4 * 1) = 150 constraints
+            for (y, col) in OFF.iter().enumerate() {
+                for (x, off) in col.iter().enumerate() {
+                    let word_e = Self::from_quarters(&self.keccak_state.pi_rho_dense_e, Some(y), x);
+                    let quo_e =
```

Comment:
Don't we have to check it is a boolean?

Reply:
Not here. Here instead, we need to check that quotient_e is <2^64.

We check it is a boolean for state_c because the rotation in step theta is just by 1 bit. That means that the quotient can only be 1 or 0.

Reply:
Ah, yes, right!